packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -768,13 +768,25 @@ class AppBar extends StatefulWidget implements PreferredSizeWidget {
 }
 
 class _AppBarState extends State<AppBar> {
+  final Map<String?, double> _scrollPositions = <String?, double>{};
   ScrollNotificationObserverState? _scrollNotificationObserver;
   bool _scrolledUnder = false;
 
   @override
   void didChangeDependencies() {
     super.didChangeDependencies();
     _scrollNotificationObserver?.removeListener(_handleScrollNotification);
+    final ScaffoldState? scaffoldState = Scaffold.maybeOf(context);
+
+    if (scaffoldState != null) {
+      final bool isDrawerOpen = scaffoldState.isDrawerOpen;
+      final bool isEndDrawerOpen = scaffoldState.hasEndDrawer && scaffoldState.isEndDrawerOpen;
+
+      if (isDrawerOpen || isEndDrawerOpen) {
+        _scrollNotificationObserver = null;
```

Comment:
Maybe this is what is causing the Google test failures. If the notifier is null, then maybe we are not updating the scrolled under state anymore in some scenarios where we should be?

Reply:
Unfortunately that is not the problem.

Reply:
Ah I see. Hm, unfortunately I don't think we can land this change since it appears to be breaking something that we have not been able to identify.

Reply:
After commenting on the section below, the Google testing passed normally. There is probably some behavior with multiple scrolls that this code breaks or is not expected.
```
if (_scrollPositions.isNotEmpty && notification.metrics.axis == Axis.vertical) {
    if (_scrollPositions.values.toList().any((double element) => element > 0) && _scrolledUnder) {
       return;
     }
}
```

Reply:
The Google testing shard is a random sampling of tests, so it is possible it did not pick up the test in question.
It looks like the framework tests are failing now though.

Reply:
I commented out the code in question, I'll revert it. If it's better, I'll leave the issue of multiple scrolls for another PR.

Reply:
I removed the check for multiple scrolls in this PR. I'll create another one so it doesn't block the main fix.